### PR TITLE
2023 03 14

### LINKED ISSUE #237: scanner overlaps other components
Screenshot_20230307-042340

worse than this on some systems

### PR DESCRIPTION
**Pull request by fluidvanadium**
in this branch, i modified the qrcodescanner to correctly display in android. unfortunately, i am unable to test ios...

also enabled the pixelometer in the dev environment only

### CODE REVIEW

**Comment by fluidvanadium** [comment]
oops, now yarn testing -u

**Comment by juanky201271** [comment]
@fluidvanadium can you added here a couple of screenshots with before & after of this PR. I'm asking because in my device I can't see the title `Scan a Zcash Address` and the `cancel` button is now inside of the camera box. I will try in an emulator as well...

In the Emulator is slightly better... (the title & the button should be vertically center in their black spaces)...

Screenshot_1678917501

**Comment by fluidvanadium** [comment]
so.... it works differently on ios and android... i may try to replace it with a newer component

**Comment by juanky201271** [comment]
> so.... it works differently on ios and android... i may try to replace it with a newer component

I didn't try yet in IOS, my testing for now is in a Android Physical device (Samsung galaxy tablet) & Android emulator (Pixel XL Android 13 SDK 33 x86_64).
But I'm sure that in Mac/ios we have to adjust something as well.

**Comment by fluidvanadium** [comment]
pre-fix on physical:

Screenshot_20230316-020639

pre-fix on emulator

Screenshot from 2023-03-16 02-02-36

post-fix on emulator

Screenshot from 2023-03-16 02-05-32

**Comment by fluidvanadium** [comment]
it's clear that whatever we have isn't exactly right
why are we subtracting and dividing by constant factors instead of having the correct bounding box

i will continue to work on it

**Comment by fluidvanadium** [comment]
rather than kludge the numbers, i would like to create the perfect bounding box for display on all devices. i believe the tools are there

**Comment by juanky201271** [comment]
take a look at this: https://github.com/react-native-camera/react-native-camera/issues/2388#issuecomment-520123596

`overflow` can be a good tool.

**Comment by fluidvanadium** [comment]
> take a look at this: [react-native-camera/react-native-camera#2388 (comment)](https://github.com/react-native-camera/react-native-camera/issues/2388#issuecomment-520123596)

your refactor really helped me! like a catapult

okay, so i got rid of most of that file. now it works great!

File: components/Send/components/ScannerAddress.tsx [hunk 1]
```diff
@@ -71,36 +69,26 @@ const ScannerAddress: React.FunctionComponent<ScannerAddressProps> = ({ updateTo
 
   const { colors } = useTheme();
   return (
-    <View style={{ width: '100%', height: '100%' }}>
+    <View style={{
```
Comment (juanky201271) [inline components/Send/components/ScannerAddress.tsx:72]: I was testing in IOS... and you need to change this `View` to `SafeAreaView` from react-native. In Android have no effect but in IOS is important.
Only the `View` more external.

**Review by juanky201271 — CHANGES_REQUESTED** [review]

File: components/Send/Send.tsx [hunk 2]
```diff
@@ -348,7 +348,7 @@ const Send: React.FunctionComponent<SendProps> = ({
     })();
   }, [isFocused]);
 
-  //console.log('render send', 'w', dimensions.width, 'h', dimensions.height);
+  console.log('render send', 'w', dimensions.width, 'h', dimensions.height);
```
Comment (juanky201271) [inline components/Send/Send.tsx:351]: and delete this log -> it's a little bit annoying... hehehe.

**Review by juanky201271 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**